### PR TITLE
cli test for deleting versions of file

### PR DESCRIPTION
**Pull request by bhawanaprasain**
## Description
Available versions of a file can be deleted using cli.

## Related Issue
#33051

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [X] Tests only (no source changes)

### CODE REVIEW

**Comment by bhawanaprasain** [comment]
@dpakach @skshetry @paurakhsharma please review on my PR.

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 1]
```diff
@@ -648,6 +648,20 @@ public function theAdministratorEmptiesTheTrashbinOfUserUsingTheOccCommand($user
 		);
 	}
 
+	/**
+	 * @When user :user deletes versions of file :file using occ comand
+	 *
+	 * @param string $user
+	 * @param string $file
```
Comment (skshetry) [inline tests/acceptance/features/bootstrap/OccContext.php]:
```suggestion

```

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 2]
```diff
@@ -648,6 +648,20 @@ public function theAdministratorEmptiesTheTrashbinOfUserUsingTheOccCommand($user
...
+	 *
+	 * @return void
+	 */
+	public function userDeletesVersionsOfFileUsingOccComand($user, $file) {
```
Comment (skshetry) [inline tests/acceptance/features/bootstrap/OccContext.php]:
```suggestion
	public function userDeletesVersionsOfFileUsingOccComand($user) {
```

File: tests/acceptance/features/cliMain/fileVersions.feature [hunk 3]
```diff
@@ -0,0 +1,14 @@
+@cli @TestAlsoOnExternalUserBackend
+
+Feature: check file versions
+
+  Scenario: user clears the versions of a file
+    And user "user0" has been created with default attributes and without skeleton files
```
Comment (skshetry) [inline tests/acceptance/features/cliMain/fileVersions.feature]:
```suggestion
    Given user "user0" has been created with default attributes and without skeleton files
```

File: tests/acceptance/features/cliMain/fileVersions.feature [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    Given user "user0" has created folder "/sharingfolder"
+    And user "user0" has uploaded file with content "old content" to "/sharingfolder/sharefile.txt"
+    And user "user0" has uploaded file with content "version 1" to "/sharingfolder/sharefile.txt"
+    And user "user0" has uploaded file with content "version 2" to "/sharingfolder/sharefile.txt"
+    And user "user0" has uploaded file with content "version 3" to "/sharingfolder/sharefile.txt"
+    And the version folder of file "sharingfolder/sharefile.txt" for user "user0" should contain "3" element
+    When user "user0" deletes versions of file "/sharingfolder/sharefile.txt" using occ comand
```
Comment (skshetry) [inline tests/acceptance/features/cliMain/fileVersions.feature]:
```suggestion
    When the administrator deletes all the versions for user "user0" using occ comand
```

File: tests/acceptance/features/cliMain/fileVersions.feature [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+Feature: check file versions
+
+  Scenario: user clears the versions of a file
+    And user "user0" has been created with default attributes and without skeleton files
+    Given user "user0" has created folder "/sharingfolder"
+    And user "user0" has uploaded file with content "old content" to "/sharingfolder/sharefile.txt"
+    And user "user0" has uploaded file with content "version 1" to "/sharingfolder/sharefile.txt"
+    And user "user0" has uploaded file with content "version 2" to "/sharingfolder/sharefile.txt"
+    And user "user0" has uploaded file with content "version 3" to "/sharingfolder/sharefile.txt"
+    And the version folder of file "sharingfolder/sharefile.txt" for user "user0" should contain "3" element
```
Comment (skshetry) [inline tests/acceptance/features/cliMain/fileVersions.feature]:
```suggestion
    Then the version folder of file "sharingfolder/sharefile.txt" for user "user0" should contain "3" element
```

**Comment by bhawanaprasain** [comment]
@dpakach @skshetry @paurakhsharma please review again.

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 6]
```diff
@@ -648,6 +648,19 @@ public function theAdministratorEmptiesTheTrashbinOfUserUsingTheOccCommand($user
 		);
 	}
 
+	/**
+	 * @When the administrator deletes all the versions for user :user0
```
Comment (paurakhsharma) [inline tests/acceptance/features/bootstrap/OccContext.php]:
```suggestion
	 * @When the administrator deletes all the versions for user :user
```

**Review by paurakhsharma — COMMENTED** [review]
Looks good to me :+1:

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
Backport `stable10` #35838

**Comment by bhawanaprasain** [comment]
@phil-davis please review again

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@bhawanaprasain please rebase.
Something happened - drone did not start.

**Comment by phil-davis** [comment]
I made a new PR #35869 because I can't find a way to rebase this one.

**Pull request closed without merging**